src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4859,13 +4859,39 @@ public PreprocessingSymbolInfo GetPreprocessingSymbolInfo(IdentifierNameSyntax n
         {
             CheckSyntaxNode(node);
 
-            if (node.Ancestors().Any(n => SyntaxFacts.IsPreprocessorDirective(n.Kind())))
```

Comment:
undo these changes if not needed.

Reply:
The changes on this line are needed to filter out unrelated directives like `#region`, etc.
In other lines, the changes are necessary to discover pp symbol names in `DefineDirectiveTriviaSyntax` and `UndefDirectiveTriviaSyntax` that contain the pp symbol name as a `SyntaxToken`, and not an `IdentifierNameSyntax`. The same applies to the changes in the VB semantic model.

Reply:
all of these then need tests at the compiler layer if you are changing the behavior of existing APIs.